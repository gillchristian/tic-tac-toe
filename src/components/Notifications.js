import React from 'react'
import PropTypes from 'prop-types'

import Alert from './styled/Alert'
import Button from './styled/Button'


const Notifications = ({ startGame, children }) => (
  <Alert>
    <span>{children}</span>
    <Button onClick={startGame}>New Game</Button>
  </Alert>
)

Notifications.propTypes = {
  startGame: PropTypes.func.isRequired,
  children: PropTypes.node
}

export default Notifications

