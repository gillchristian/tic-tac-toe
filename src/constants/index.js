export const O = '●'
export const X = '✖'
export const s = ''

export const RESULTS = [
  /1..1..1../,
  /.1..1..1./,
  /..1..1..1/,
  /111....../,
  /...111.../,
  /......111/,
  /1...1...1/,
  /..1.1.1../
]

