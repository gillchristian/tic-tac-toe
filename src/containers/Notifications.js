import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

import Notifications from '../components/Notifications'
import EndingMessage from '../components/EndingMessage'

import { startGame } from '../actions/game'
import { isOver } from '../reducers/is-over'
import { getWinner } from '../reducers/winner'

const mapStateToProps = (state) => ({
  isOver: isOver(state),
  winner: getWinner(state)
})

const mapDispatchToProps = {
  startGame
}

const NotificationsContainer = ({ isOver, startGame, winner }) => isOver
  ? <Notifications startGame={startGame}>
      <EndingMessage>{winner}</EndingMessage>
    </Notifications>
  : null

NotificationsContainer.propTypes = {
  isOver: PropTypes.bool,
  startGame: PropTypes.func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NotificationsContainer)

