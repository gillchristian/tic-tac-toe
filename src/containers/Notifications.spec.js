import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { X, O } from '../constants'
import { NotificationsContainer } from './Notifications'
import Notifications from '../components/Notifications'

describe('Notifications container', () => {
  it('Renders the Notifications component when the game is over, or null if not over', () => {
    const props = {
      startGame: () => {},
      isOver: true,
      winner: X
    }
    let wrapper = shallow(<NotificationsContainer {...props} />)

    expect(toJson(wrapper)).toMatchSnapshot()

    expect(wrapper.type()).toEqual(Notifications)

    props.isOver = false

    wrapper = shallow(<NotificationsContainer {...props} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

