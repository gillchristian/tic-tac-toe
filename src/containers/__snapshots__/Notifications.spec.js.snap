// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications container Renders the Notifications component when the game is over, or null if not over 1`] = `
<Notifications
  startGame={[Function]}
>
  <Message>
    X
  </Message>
</Notifications>
`;

exports[`Notifications container Renders the Notifications component when the game is over, or null if not over 2`] = `null`;
